operators/pkg/tenant-controller/nextcloud_utils.go

```diff
@@ -0,0 +1,173 @@
+package tenant_controller
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	resty "github.com/go-resty/resty/v2"
+	"k8s.io/klog/v2"
+)
+
+// NcHandler defines the method need to interact with nextcloud
+type NcHandler interface {
+	GetUser(ncUsername string) (found bool, displayname *string, err error)
+	CreateUser(ncUsername, ncPsw, displayname string) error
+	UpdateUserData(username, param, value string) error
+	DeleteUser(username string) error
+}
+
+// NcActor holds the info and methods to interact with nextcloud
+type NcActor struct {
+	Client   *resty.Client
+	TnOpUser string
+	TnOpPsw  string
```

Comment:
Suppose should be Nc instead of Tn

Reply:
I used `TnOp` because in theory I imagined an account for the tenant operator to act on nextcloud, a bit like the one in keycloak

Reply:
Gotcha